src/testrender/background.h

```diff
@@ -10,17 +10,49 @@
 
 OSL_NAMESPACE_ENTER
 
+#ifdef __CUDACC__
+// std::upper_bound is not supported in device code, so define a version of it here.
+// Adapted from the LLVM Project, see https://llvm.org/LICENSE.txt for license information.
+template<typename T>
+inline OSL_HOSTDEVICE const T*
+upper_bound_cuda(const T* data, int count, const T value)
```

Comment:
Many times linear search will  be faster on gpu :)

Reply:
In my experience not really, though it would be worth benchmarking. Also I believe this is used for the background pdf which can be a few thousand elements long.